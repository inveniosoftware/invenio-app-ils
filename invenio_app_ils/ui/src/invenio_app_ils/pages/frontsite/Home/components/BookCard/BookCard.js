import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Card, Image } from 'semantic-ui-react';

export default class BookCard extends Component {
  render() {
    const bookData = this.props.bookData;
    return (
      <Card onClick={bookData.onClick} data-test={bookData.pid}>
        <Image size={bookData.imageSize} src={bookData.imageCover} />
        <Card.Content>
          <Card.Header>{bookData.title}</Card.Header>
          <Card.Meta>{bookData.authors}</Card.Meta>
        </Card.Content>
      </Card>
    );
  }
}

BookCard.propTypes = {
  bookData: PropTypes.object.isRequired,
};
