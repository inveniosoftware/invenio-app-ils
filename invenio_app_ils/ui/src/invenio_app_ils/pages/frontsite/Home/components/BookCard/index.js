export { default as BookCard } from './BookCard';
