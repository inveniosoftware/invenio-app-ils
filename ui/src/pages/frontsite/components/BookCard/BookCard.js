import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Card, Image, Label } from 'semantic-ui-react';
import isEmpty from 'lodash/isEmpty';
import { FrontSiteRoutes } from '@routes/urls';
import { getCover } from '../../config';
import { goTo } from '@history';
import { DocumentAuthors } from '@components/Document';

export class BookCard extends Component {
  renderLabels = meta => {
    return (
      <>
        {meta.circulation.has_items_for_loan > 0 && <Label>On shelf</Label>}
        {meta.eitems.total > 0 && <Label>E-book</Label>}
      </>
    );
  };

  render() {
    const { data } = this.props;
    return (
      <Card
        link
        centered
        className={'fs-book-card'}
        onClick={() =>
          goTo(FrontSiteRoutes.documentDetailsFor(data.metadata.pid))
        }
        data-test={data.metadata.pid}
      >
        <Card.Meta className={'discrete'}>
          {data.metadata.document_type}
        </Card.Meta>
        <Image
          centered
          src={getCover(data.metadata.edition)}
          size={'small'}
          onError={e => (e.target.style.display = 'none')}
        />
        <Card.Content>
          <Card.Header>{data.metadata.title}</Card.Header>
          <Card.Meta>
            <DocumentAuthors metadata={data.metadata} />
            <div>
              {!isEmpty(data.metadata.imprints) ? (
                <>
                  {data.metadata.imprints[0].date} <br />{' '}
                </>
              ) : null}
              Edition {data.metadata.edition}
            </div>
          </Card.Meta>
        </Card.Content>
        <Card.Content extra>{this.renderLabels(data.metadata)}</Card.Content>
      </Card>
    );
  }
}

BookCard.propTypes = {
  data: PropTypes.object.isRequired,
};
