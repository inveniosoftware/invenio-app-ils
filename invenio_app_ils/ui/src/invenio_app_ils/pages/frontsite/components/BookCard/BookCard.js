import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Card, Image, Label } from 'semantic-ui-react';
import { FrontSiteRoutes } from '../../../../routes/urls';
import { getCover } from '../../config';
import { goTo } from '../../../../history';
import { DocumentAuthors } from '../../Documents/DocumentsDetails/DocumentMetadata/components';

export class BookCard extends Component {
  renderLabels = meta => {
    return (
      <>
        {meta.circulation.has_items_for_loan > 0 && <Label>On shelf</Label>}
        {meta.eitems.total > 0 && <Label>E-book</Label>}
      </>
    );
  };

  render() {
    const { data } = this.props;
    return (
      <Card
        link
        centered
        className={'fs-book-card'}
        onClick={() =>
          goTo(FrontSiteRoutes.documentDetailsFor(data.metadata.pid))
        }
        data-test={data.metadata.pid}
      >
        <Image
          centered
          src={getCover(data.metadata.pid)}
          size={'small'}
          onError={e => (e.target.style.display = 'none')}
        />
        <Card.Content centered={'true'}>
          <Card.Header>{data.metadata.title}</Card.Header>
          <Card.Meta>
            <DocumentAuthors metadata={data.metadata} />
          </Card.Meta>
          {this.renderLabels(data.metadata)}
        </Card.Content>
      </Card>
    );
  }
}

BookCard.propTypes = {
  data: PropTypes.object.isRequired,
};
