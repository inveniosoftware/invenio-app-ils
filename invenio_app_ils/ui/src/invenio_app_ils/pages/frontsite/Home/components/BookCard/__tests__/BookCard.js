import React from 'react';
import { mount } from 'enzyme';
import BookCard from '../BookCard';

describe('BookCard tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const bookData = {
    pid: '123',
    title: 'Lorem',
    authors: 'Author1, Author2',
    imageSize: 'small',
    imageCover: '',
    eitems: [],
    onClick: jest.fn(),
  };

  it('should render the document correctly', () => {
    component = mount(<BookCard bookData={bookData} />);
    expect(component).toMatchSnapshot();

    const rows = component
      .find('BookCard')
      .find('Card')
      .filterWhere(element => element.prop('data-test') === bookData.pid);
    expect(rows).toHaveLength(1);
  });
});
