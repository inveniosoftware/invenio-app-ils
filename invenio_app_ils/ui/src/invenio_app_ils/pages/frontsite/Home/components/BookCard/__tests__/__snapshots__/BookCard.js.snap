// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard tests should render the document correctly 1`] = `
<BookCard
  bookData={
    Object {
      "authors": "Author1, Author2",
      "eitems": Array [],
      "imageCover": "",
      "imageSize": "small",
      "onClick": [MockFunction],
      "pid": "123",
      "title": "Lorem",
    }
  }
>
  <Card
    data-test="123"
    onClick={[MockFunction]}
  >
    <a
      className="ui card"
      data-test="123"
      onClick={[Function]}
    >
      <Image
        as="img"
        size="small"
        src=""
        ui={true}
      >
        <img
          className="ui small image"
          src=""
        />
      </Image>
      <CardContent>
        <div
          className="content"
        >
          <CardHeader>
            <div
              className="header"
            >
              Lorem
            </div>
          </CardHeader>
          <CardMeta>
            <div
              className="meta"
            >
              Author1, Author2
            </div>
          </CardMeta>
        </div>
      </CardContent>
    </a>
  </Card>
</BookCard>
`;
